src/SeleniumLibrary/keywords/alert.py

```diff
@@ -14,153 +14,183 @@
+        """Deprecated. Use `Input Text Into Alert` instead.
 
-    def __init__(self, ctx):
-        LibraryComponent.__init__(self, ctx)
-        self._next_alert_dismiss_type = self.ACCEPT_ALERT
+        Types the given ``text`` into an input field in an alert.
+        Leaves the alert open.
+        """
+        self.input_text_into_alert(text, self.LEAVE)
 
     @keyword
-    def input_text_into_prompt(self, text):
-        """Types the given `text` into alert box.  """
-        try:
-            alert = self._wait_alert()
-            alert.send_keys(text)
-        except WebDriverException:
-            raise RuntimeError('There were no alerts')
+    def input_text_into_alert(self, text, action=ACCEPT):
+        """Types the given ``text`` into an input field in an alert.
+
+        The alert is accepted by default, but that behavior can be controlled
+        by using the ``action`` argument same way as with `Handle Alert`.
+
+        New in SeleniumLibrary 3.0.
+        """
+        alert = self._wait_alert()
+        alert.send_keys(text)
+        self._handle_alert(alert, action)
+
+    @keyword
+    def alert_should_be_present(self, text='', action=ACCEPT):
+        """Verifies that an alert is present and, by default, accepts it.
+
+        Fails if no alert is present. If ``text`` is a non-empty string,
+        then it is used to verify alert's message. The alert is accepted
+        by default, but that behavior can be controlled by using the
+        ``action`` argument same way as with `Handle Alert`.
+
+        The ``action`` argument is new in SeleniumLibrary 3.0. In earlier
+        versions the alert was always accepted.
+        """
+        message = self.handle_alert(action)
+        if text and text != message:
+            raise AssertionError("Alert message should have been '%s' but it "
+                                 "was '%s'." % (text, message))
 
     @keyword
-    def alert_should_be_present(self, text=''):
-        """Verifies an alert is present and dismisses it.
+    def alert_should_not_be_present(self, action=ACCEPT):
+        """Verifies that no alert is present.
 
-        If `text` is a non-empty string, then it is also verified that the
-        message of the alert equals to `text`.
+        If the alert actually exists, ``action`` argument determines
+        how it should be handled same way as with `Handle Alert`.
 
-        Will fail if no alert is present. Note that following keywords
-        will fail unless the alert is dismissed by this
-        keyword or another like `Get Alert Message`.
+        New in SeleniumLibrary 3.0.
         """
-        alert_text = self._handle_alert(self.ACCEPT_ALERT)
-        if text and alert_text != text:
-            raise AssertionError("Alert text should have been "
-                                 "'%s' but was '%s'"
-                                 % (text, alert_text))
+        try:
+            alert = self._wait_alert(timeout=0)
```

Comment:
Why the timeout is zero? It would make the tests more stable if we allow small polling.

Reply:
We expect that there's no alert so in the common case *any* polling would slow down execution unnecessarily. Very small timeout could be considered, but at least using the global timeout that's 5 seconds by default would be a very bad idea. I'd leave the default timeout here to be zero and just make the timeout configurable with an optional argument.